### ISSUE TITLE
throw `Error: ENOENT: no such file or directory, stat '...'` when using Vite

### ISSUE THREAD

**Issue by johnsoncodehk** [timostamm/protobuf-ts#38]
First thanks for your great work, it is very easy to use!

My only problem is when I start a Vite server, there will throw many `Error: ENOENT: no such file or directory, stat '...'` errors.

I found the reason is because the npm publish has include `.map` files and the mapping `.ts` files do not include.

I see you have add `.js.map` to `.gitignore` now, but not enough. if `"sourceMap": true` is set, there will add a line like `//# sourceMappingURL=assert.js.map` to `.js` files, so there still has `.map` files not found errors, you must to disable sourceMap option for publish.

**Comment by timostamm** [comment]
Thanks for the feedback, @johnsoncodehk 

I did a `npm install @protobuf-ts/runtime` in a fresh new project and had a look at the contents of `node_modules/@protobuf-ts/runtime/`. 

![Bildschirmfoto 2020-11-26 um 11 15 31](https://user-images.githubusercontent.com/4289451/100338340-c0dbbe00-2fd8-11eb-9c06-36f52d1cf3a7.png)

(I removed all source files except "assert") just to make it a bit easier. 

Looking at `build/commonjs/assert.js`:

```javascript
// code omitted
//# sourceMappingURL=assert.js.map
```

`build/es2015/assert.js`:

```javascript
// code omitted
//# sourceMappingURL=assert.js.map
```

`build/types/assert.d.ts`:

```typescript
// declarations omitted
//# sourceMappingURL=assert.d.ts.map
```

All files referenced by `sourceMappingURL` exist. The mapping files look very similar. This is `build/commonjs/assert.js.map`: 

```
{"version":3,"file":"assert.js","sourceRoot":"","sources":["../../src/assert.ts"],"names":[],"mappings":";;;AAAA;;GAEG;AACH,SAAgB,MAAM,CAAC,SAAc,EAAE,GAAY;IAC/C,IAAI,CAAC,SAAS,EAAE;QACZ,MAAM,IAAI,KAAK,CAAC,GAAG,CAAC,CAAC;KACxB;AACL,CAAC;AAJD,wBAIC;AAED;;GAEG;AACH,SAAgB,WAAW,CAAC,KAAY,EAAE,GAAY;IAClD,MAAM,IAAI,KAAK,CAAC,GAAG,aAAH,GAAG,cAAH,GAAG,GAAI,qBAAqB,GAAG,KAAK,CAAC,CAAC;AAC1D,CAAC;AAFD,kCAEC;AAGD,MACI,WAAW,GAAI,sBAAsB,EACrC,WAAW,GAAG,CAAC,sBAAsB,EACrC,UAAU,GAAG,UAAU,EACvB,SAAS,GAAG,UAAU,EACtB,SAAS,GAAG,CAAC,UAAU,CAAC;AAG5B,SAAgB,WAAW,CAAC,GAAQ;IAChC,IAAI,OAAO,GAAG,KAAK,QAAQ;QACvB,MAAM,IAAI,KAAK,CAAC,kBAAkB,GAAG,OAAO,GAAG,CAAC,CAAC;IACrD,IAAI,CAAC,MAAM,CAAC,SAAS,CAAC,GAAG,CAAC,IAAI,GAAG,GAAG,SAAS,IAAI,GAAG,GAAG,SAAS;QAC5D,MAAM,IAAI,KAAK,CAAC,kBAAkB,GAAG,GAAG,CAAC,CAAC;AAClD,CAAC;AALD,kCAKC;AAED,SAAgB,YAAY,CAAC,GAAQ;IACjC,IAAI,OAAO,GAAG,KAAK,QAAQ;QACvB,MAAM,IAAI,KAAK,CAAC,mBAAmB,GAAG,OAAO,GAAG,CAAC,CAAC;IACtD,IAAI,CAAC,MAAM,CAAC,SAAS,CAAC,GAAG,CAAC,IAAI,GAAG,GAAG,UAAU,IAAI,GAAG,GAAG,CAAC;QACrD,MAAM,IAAI,KAAK,CAAC,mBAAmB,GAAG,GAAG,CAAC,CAAC;AACnD,CAAC;AALD,oCAKC;AAED,SAAgB,aAAa,CAAC,GAAQ;IAClC,IAAI,OAAO,GAAG,KAAK,QAAQ;QACvB,MAAM,IAAI,KAAK,CAAC,oBAAoB,GAAG,OAAO,GAAG,CAAC,CAAC;IACvD,IAAI,CAAC,MAAM,CAAC,QAAQ,CAAC,GAAG,CAAC;QACrB,OAAO;IACX,IAAI,GAAG,GAAG,WAAW,IAAI,GAAG,GAAG,WAAW;QACtC,MAAM,IAAI,KAAK,CAAC,oBAAoB,GAAG,GAAG,CAAC,CAAC;AACpD,CAAC;AAPD,sCAOC"}
```

So the problem seems to be `../../src/assert.ts`. This path simply does not exists in the published package...

Could you verify if your error goes away if you add the .ts source files?  
You can download them from here: https://github.com/timostamm/protobuf-ts/tree/v1.x/packages/runtime/src (note the v1.x branch)  
Then place the ts files in your `node_modules/@protobuf-ts/runtime/src/` directory. 

If this fixes your error, I will add the files to the published packages.

**Comment by johnsoncodehk** [comment]
Yes current release version have `.map` files, but I see you have added `.js.map` in `.gitignore`: https://github.com/timostamm/protobuf-ts/blob/7f3cfebadb6f971fb81aa1bafd35837fe67c0917/packages/runtime/.gitignore#L6

so I guess `.map` will remove in next version, maybe I'm wrong.

I already remove `.map` and `//# sourceMappingURL=` in my local node_modules to remove errors, so you don't need to publish version for me now :).

Yes if you add `.ts` files to npm release can resolve this problem, but I think disable sourceMap for publish build is better, hope it can fix in next version.

And thanks you work again!

**Comment by timostamm** [comment]
> I see you have added `.js.map` in `.gitignore`

Yes, but there also is a [.npmignore](https://github.com/timostamm/protobuf-ts/blob/7f3cfebadb6f971fb81aa1bafd35837fe67c0917/packages/runtime/.npmignore). This means .gitignore does not matter for npm publish. 


> I think disable sourceMap for publish build is better

Seems like there is no definite answer, see googleapis/google-cloud-node#2867 for example. 

But I think I prefer removing source maps instead of adding typescript sources. I'll give it some more thought. Let's keep this issue open as a reminder.

**Comment by johnsoncodehk** [comment]
Yes I think removing source maps is better too! (may be some misunderstanding before)

**Comment by timostamm** [comment]
@johnsoncodehk, source mappings have been removed in prerelease version v2.0.0-alpha.2. 

If this does not fix the issue, please reopen.

**Comment by timostamm** [comment]
The fix has been back-ported in v1.0.12

**Comment by johnsoncodehk** [comment]
I already using 2.x, thanks!
